db/tag.c

```diff
@@ -6750,7 +6750,14 @@ void freedb_int(dbtable *db, dbtable *replace)
     }
 
     if (replace) {
+        pthread_rwlock_t lk_copy;
+        // Make a copy of original sc_live_lk
+        memcpy(&lk_copy, &db->sc_live_lk, sizeof(pthread_rwlock_t));
+
         memcpy(db, replace, sizeof(dbtable));
+
+        // Restore sc_live_lk
+        memcpy(&db->sc_live_lk, &lk_copy, sizeof(pthread_rwlock_t));
         db->dbs_idx = dbs_idx;
     } else
         free(db);
```

Comment:
I don't think it's defined behavior to copy a mutex, with memcpy or an assignment.  We've hopefully already done a pthread_rwlock_destroy on the destination, we should pthread_rwlock_init on it after the copy.